File: examples/javascript/validator.js
const { Validator } = require("./binary-options-tools.node");

// Create validator instances
const none = new Validator();
const regex = Validator.regex(/([A-Z])\w+/);
const start = Validator.startsWith("Hello");
const end = Validator.endsWith("Bye");
const contains = Validator.contains("World");
const rnot = Validator.not(contains);

// Modified for better testing - smaller groups with predictable outcomes
const rall = Validator.all([regex, start]); // Will need both capital letter and "Hello" at start
const rany = Validator.any([contains, end]); // Will need either "World" or end with "Bye"

// Testing each validator
console.log(`None validator: ${none.check("hello")} (Expected: true)`);

console.log(`Regex validator: ${regex.check("Hello")} (Expected: true)`);
console.log(`Regex validator: ${regex.check("hello")} (Expected: false)`);

console.log(
  `Starts_with validator: ${start.check("Hello World")} (Expected: true)`,
);
console.log(
  `Starts_with validator: ${start.check("hi World")} (Expected: false)`,
);

console.log(`Ends_with validator: ${end.check("Hello Bye")} (Expected: true)`);
console.log(
  `Ends_with validator: ${end.check("Hello there")} (Expected: false)`,
);

console.log(
  `Contains validator: ${contains.check("Hello World")} (Expected: true)`,
);
console.log(
  `Contains validator: ${contains.check("Hello there")} (Expected: false)`,
);

console.log(`Not validator: ${rnot.check("Hello World")} (Expected: false)`);
console.log(`Not validator: ${rnot.check("Hello there")} (Expected: true)`);

// Testing the all validator
console.log(`All validator: ${rall.check("Hello World")} (Expected: true)`); // Starts with "Hello" and has capital
console.log(`All validator: ${rall.check("hello World")} (Expected: false)`); // No capital at start
console.log(`All validator: ${rall.check("Hey there")} (Expected: false)`); // Has capital but doesn't start with "Hello"

// Testing the any validator
console.log(`Any validator: ${rany.check("Hello World")} (Expected: true)`); // Contains "World"
console.log(`Any validator: ${rany.check("Hello Bye")} (Expected: true)`); // Ends with "Bye"
console.log(`Any validator: ${rany.check("Hello there")} (Expected: false)`); // Neither contains "World" nor ends with "Bye"

File: examples/python/sync/validator.py
# Same thing as in the async folder, the code is sync and can be passed to both the async and sync implementations of the create_raw_order* methods.

from BinaryOptionsToolsV2.validator import Validator

if __name__ == "__main__":
    none = Validator()
    regex = Validator.regex("([A-Z])\w+")
    start = Validator.starts_with("Hello")
    end = Validator.ends_with("Bye")
    contains = Validator.contains("World")
    rnot = Validator.ne(contains)
    custom = Validator.custom(lambda x: x.startswith("Hello") and x.endswith("World"))

    # Modified for better testing - smaller groups with predictable outcomes
    rall = Validator.all(
        [regex, start]
    )  # Will need both capital letter and "Hello" at start
    rany = Validator.any([contains, end])  # Will need either "World" or end with "Bye"

    print(f"None validator: {none.check('hello')} (Expected: True)")
    print(f"Regex validator: {regex.check('Hello')} (Expected: True)")
    print(f"Regex validator: {regex.check('hello')} (Expected: False)")
    print(f"Starts_with validator: {start.check('Hello World')} (Expected: True)")
    print(f"Starts_with validator: {start.check('hi World')} (Expected: False)")
    print(f"Ends_with validator: {end.check('Hello Bye')} (Expected: True)")
    print(f"Ends_with validator: {end.check('Hello there')} (Expected: False)")
    print(f"Contains validator: {contains.check('Hello World')} (Expected: True)")
    print(f"Contains validator: {contains.check('Hello there')} (Expected: False)")
    print(f"Not validator: {rnot.check('Hello World')} (Expected: False)")
    print(f"Not validator: {rnot.check('Hello there')} (Expected: True)")
    try:
        print(f"Custom validator: {custom.check('Hello World')}, (Expected: True)")
        print(f"Custom validator: {custom.check('Hello there')}, (Expected: False)")
    except Exception as e:
        print(f"Error: {e}")
    # Testing the all validator
    print(
        f"All validator: {rall.check('Hello World')} (Expected: True)"
    )  # Starts with "Hello" and has capital
    print(
        f"All validator: {rall.check('hello World')} (Expected: False)"
    )  # No capital at start
    print(
        f"All validator: {rall.check('Hey there')} (Expected: False)"
    )  # Has capital but doesn't start with "Hello"

    # Testing the any validator
    print(
        f"Any validator: {rany.check('Hello World')} (Expected: True)"
    )  # Contains "World"
    print(
        f"Any validator: {rany.check('Hello Bye')} (Expected: True)"
    )  # Ends with "Bye"
    print(
        f"Any validator: {rany.check('Hello there')} (Expected: False)"
    )  # Neither contains "World" nor ends with "Bye"
